Dockerfile

```
@@ -2,8 +2,16 @@ FROM registry.ci.openshift.org/openshift/release:golang-1.19 AS builder
 
 WORKDIR /github.com/redhat-appstudio/e2e-tests
 USER root
-COPY . .
-RUN GOOS=linux make build
+
+COPY cmd/ cmd/
+COPY magefiles/ magefiles/
+COPY pkg/ pkg/
+COPY tests/ tests/
+COPY Makefile .
+COPY go.mod .
+COPY go.sum .
```

I do not know if building the image over and over locally is a use case but if it is, if you do this instead, then the downloaded modules will be cached in a layer and layer will only be invalidated if mod files are changed:

```suggestion
COPY go.mod .
COPY go.sum .
RUN go mod download -x
COPY cmd/ cmd/
COPY magefiles/ magefiles/
COPY pkg/ pkg/
COPY tests/ tests/
COPY Makefile .
```

---

it's not a typical use case, but it is handy, thanks!